test/unit/spec/util/networkmonitor.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const NetworkMonitor = require('../../../../lib/util/networkmonitor');
+
+class MockConnection {
```

Comment:
@PikaJoyce You can extend this class from `EventTarget` in `lib/eventtarget.js`. Then, you don't need to implement the listener methods here. Also, can you move them to `test/lib`?